ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -749,57 +757,35 @@ public IXLTable InsertTable(DataTable data, String tableName, Boolean createTabl
                 return null;
 
             if (XLHelper.IsValidA1Address(tableName) || XLHelper.IsValidRCAddress(tableName))
-                throw new InvalidOperationException(string.Format("Table name cannot be a valid Cell Address '{0}'.", tableName));
+                throw new InvalidOperationException($"Table name cannot be a valid Cell Address '{tableName}'.");
 
             if (createTable && this.Worksheet.Tables.Any(t => t.Contains(this)))
-                throw new InvalidOperationException(String.Format("This cell '{0}' is already part of a table.", this.Address.ToString()));
+                throw new InvalidOperationException($"This cell '{this.Address.ToString()}' is already part of a table.");
 
-            if (data.Rows.Cast<DataRow>().Any())
-                return InsertTable(data.Rows.Cast<DataRow>(), tableName, createTable);
-
-            var co = _columnNumber;
-
-            foreach (DataColumn col in data.Columns)
-            {
-                Worksheet.SetValue(col.ColumnName, _rowNumber, co);
-                co++;
+            var reader = InsertDataReaderFactory.Instance.CreateReader(data);
+            return InsertTableInternal(reader, tableName, createTable, addHeadings: true, transpose: false);
             }
-
-            ClearMerged();
```

Comment:
I can't see whether this commit retains the `ClearMerged()` behaviour.

Reply:
Oh... the `.Clear()` in https://github.com/ClosedXML/ClosedXML/pull/1389/commits/1f4f3e017bba03e4fe6f658fe28ee9fc863bdf22#diff-3b8a562c15b0aca901df66d96df87e10R839 takes care of that?

Reply:
Yes